src/litserve/server.py

```diff
@@ -344,10 +380,24 @@ def api_key_auth(x_api_key: str = Depends(APIKeyHeader(name="X-API-Key"))):
         )
 
 
-def cleanup(request_buffer, uid):
-    logger.debug("Cleaning up request uid=%s", uid)
-    with contextlib.suppress(KeyError):
-        request_buffer.pop(uid)
+async def response_queue_to_buffer(
+    response_queue: mp.Queue, buffer: Dict[str, Tuple[asyncio.Queue, asyncio.Event]], stream: bool
+):
+    while True:
+        try:
+            uid, payload = response_queue.get_nowait()
+        except Empty:
+            await asyncio.sleep(0.0001)
+            continue
+
+        if stream:
```

Comment:
please condition `if stream` outside `while`, and have two different `while loops`

otherwise we keep on evaluating `if stream` but there's really no need to